.github/workflows/helm-end-to-end-tfc.yaml

```diff
@@ -0,0 +1,90 @@
+name: E2E on Terraform Cloud [Helm]
+
+on:
+  schedule:
+    - cron: '30 6 * * *'
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+
+env:
+  USE_EXISTING_CLUSTER: true
+  CLUSTER_NAME: 'this'
+  DOCKER_IMAGE: 'this'
+  KUBECONFIG: ${{ github.workspace }}/.kube/config
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@24cb9080177205b6e8c946b17badbe402adc938f # v3.4.0
+        with:
+          fetch-depth: 0
+
+      - name: Set up Go
+        uses: actions/setup-go@4d34df0c2316fe8122ab82dc22947d607c0c91f9 # v4.0.0
+        with:
+          go-version-file: 'go.mod'
+      
+      - name: Set up kind
+        uses: helm/kind-action@d8ccf8fb623ce1bb360ae2f45f323d9d5c5e9f00 # v1.5.0
+        with:
+          wait: 2m
+          version: v0.18.0 # Installs Kubernetes v1.26.3
```

Comment:
Not crucial, but we could also make this an input with a default value, in case there will be situations where we might need to test certain PRs against a specific (different) cluster version.

Reply:
As for now, this PR aims to close some testing gaps we have. However, I have some other thoughts on how to improve the process. One of them is to set the kind version up via the repo env variable and have the cluster version as an input. In this case, we have more flexibility and don't need to raise a PR to bump up the kind version, instead, we can change the env variable. The kind team keeps backward compatibility with at least 6 latest minor Kubernetes versions. Plus tons of other improvements we can discuss later. :-)